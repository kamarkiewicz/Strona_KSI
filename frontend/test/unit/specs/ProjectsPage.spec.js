import { shallowMount, mount } from '@vue/test-utils'
import ProjectsPage from '~/pages/projects/index.vue'

describe('ProjectsPage', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(ProjectsPage, {
      mocks: {
        $t: (label) => { return label },
        $d: (label) => { return label },
        localePath: (args) => { return JSON.stringify(args) },
      },
      computed: {
        entries: () => ([
          {
            id: 1,
            title: 'Drukarka 3D',
            excerpt: 'Krótki opis wprowadzający do projektu. Pełny opis projektu wraz z galerią z projektu powinien być raczej na podstronie.',
            icon: 'fa-print',
          },
          {
            id: 2,
            title: 'Strona_KSI.',
            excerpt: 'Jest to projekt zrodzony z potrzeby nowoczesnej, dostępnej na urządzeniach mobilnych wizytówce w sieci.',
            icon: 'fa-code',
          },
          {
            id: 3,
            title: 'S.O.F.A.',
            excerpt: 'Sofa na kółkach - korzystając z doświadczenia zebranego przy łaziku wysyłamy odrobinę luksusu na Księżyc!',
            icon: 'fa-microchip',
          },
          {
            id: 4,
            title: 'Elektroniczny zamek',
            excerpt: 'Nigdy więcej zgubionego kluczyka do pokoju koła ;)',
            icon: 'fa-key',
          },
        ])
      },
    })
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
