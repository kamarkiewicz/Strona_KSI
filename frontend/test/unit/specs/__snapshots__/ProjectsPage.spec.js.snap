// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsPage renders correctly 1`] = `
<div class="projects-page pt-5 pb-slant">
  <div class="container">
    <header>
      <h2 class="text-primary pb-4">projects.title</h2>
      <p class="text-secondary">projects.description</p>
    </header>
  </div>
  <div class="container pb-4">
    <div class="row">
      <div class="py-5 col-md-6">
        <div class="row">
          <div class="text-center col-4">
            <i class="d-block mx-auto fa fa-5x fa-print"></i>
          </div>
          <div class="col-8">
            <h5 class="mb-3 text-primary font-weight-bold">Drukarka 3D</h5>
            <p class="my-1">Krótki opis wprowadzający do projektu. Pełny opis projektu wraz z galerią z projektu powinien być raczej na podstronie.</p>
          </div>
        </div>
      </div>
      <div class="py-5 col-md-6">
        <div class="row">
          <div class="text-center col-4">
            <i class="d-block mx-auto fa fa-5x fa-code"></i>
          </div>
          <div class="col-8">
            <h5 class="mb-3 text-primary font-weight-bold">Strona_KSI.</h5>
            <p class="my-1">Jest to projekt zrodzony z potrzeby nowoczesnej, dostępnej na urządzeniach mobilnych wizytówce w sieci.</p>
          </div>
        </div>
      </div>
    </div>
    <div class="row">
      <div class="py-5 col-md-6">
        <div class="row">
          <div class="text-center col-4">
            <i class="d-block mx-auto fa fa-5x fa-microchip"></i>
          </div>
          <div class="col-8">
            <h5 class="mb-3 text-primary font-weight-bold">S.O.F.A.</h5>
            <p class="my-1">Sofa na kółkach - korzystając z doświadczenia zebranego przy łaziku wysyłamy odrobinę luksusu na Księżyc!</p>
          </div>
        </div>
      </div>
      <div class="py-5 col-md-6">
        <div class="row">
          <div class="text-center col-4">
            <i class="d-block mx-auto fa fa-5x fa-key"></i>
          </div>
          <div class="col-8">
            <h5 class="mb-3 text-primary font-weight-bold">Elektroniczny zamek</h5>
            <p class="my-1">Nigdy więcej zgubionego kluczyka do pokoju koła ;)</p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
